deep-security-agent/tasks/install-win.yml

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018, Trend Micro
+#
+# License as per [repo](master/LICENSE)
+#
+# *********************************************************************
+# * Install the Deep Security Agent (For Windows only)
+# *********************************************************************
+---
+# *********************************************************************
+# * Detect the platform architecture
+# *********************************************************************
+- name: Setting facts (Win64)
+  set_fact:
+    agent_relative_path: software/agent/Windows/x86_64
+    agent_local_path: "{{ ansible_env.TMP }}\\ds_agent.msi"
+  when: >
+        ansible_os_family == "Windows" and
+        ansible_architecture == "64-bit"
+
+- name: Setting facts (Win32)
+  set_fact:
+    agent_relative_path: software/agent/Windows/i386
+    agent_local_path: "{{ ansible_env.TMP }}\\ds_agent.msi"
+  when: >
+        ansible_os_family == "Windows" and
+        ansible_architecture == "32-bit"
+
+- name: Make sure path is set properly
+  debug: 
+    msg: 
+      - "agent_local_path: {{ agent_local_path }}"
+      - "agent_relative_path: {{ agent_relative_path }}"
+
+# *********************************************************************
+# * Download Deep Security agent from Deep Security manager
+# *********************************************************************
+- debug:
+    msg: "Download from: https://{{ dsa_download_hostname }}:{{ dsa_download_port }}/{{ agent_relative_path }}/"
+
+- name: Download the Deep Security agent (Windows)
+  win_get_url:
+    url: "https://{{ dsa_download_hostname }}:{{ dsa_download_port }}/{{ agent_relative_path }}/" 
+    dest: "{{ agent_local_path }}"
+    validate_certs: no
+    timeout: 300
+    force: yes
+
+- win_stat:
+    path: "{{ agent_local_path }}"
+  register: agent_file
+
+- name: Check file size
+  debug:
+    msg: "Downloaded DSA installer file size : {{ agent_file.stat.size }} bytes"
+
+- fail:
+    msg: 
+      - "Downloaded file size is not valid. There are two possiblities : 1. The OS platform is not supported. 2. Deep Security Manager don't provide agent software package installer for your OS platform."
+      - "Please verify that agent software package for your platform is available using this reference article https://help.deepsecurity.trendmicro.com/10/0/Get-Started/Install/import-agent-software.html?redirected=true&Highlight=download"
+  when: agent_file.stat.size == 0
```

Comment:
Same as linux

Reply:
Please refer to install-linux.yml's comment.